tests/rainbowGrades/test_csci1100.py

```diff
@@ -0,0 +1,295 @@
+"""
+import sys
+import os
+import tempfile
+import shutil
+import subprocess
+
+
+def error_and_cleanup(tmp_path, message, error=-1):
+    """Used to exit and remove the top-level Rainbow Grades test in tmp.
+
+    :param tmp_path: The path of the temporary directory used for testing
+    :param message: An error message to be written to the terminal
+    :param error: Which error code to return (default -1)
+    :return: None
+    """
+    print(message)
+    if os.path.exists(tmp_path):
+        shutil.rmtree(tmp_path)
+    sys.exit(error)
+
+
+def remove_extra_raw_data_fields(raw_line):
+    """Used for filter() to ignore time-sensitive lines in report JSON files.
+
+    :param raw_line: A line to filter
+    :return: True if no time-sensitive fields are present in the line
+    """
+    if 'grade_released_date' in raw_line:
+        return False
+    if 'last_update' in raw_line:
+        return False
+    return True
+
+
+def remove_info_last_updated(raw_line):
+    """Used for filter() to ignore time-sensitive lines in summary HTML files.
+
+    :param raw_line: A line to filter
+    :return: True if no time-sensitive fields are present in the line
+    """
+    if '<em>Information last updated:' in raw_line:
+        return False
+    return True
+
+
+def csci1100_rainbow_grades_test():
+    """Function wrapper for the body of the test."""
+
+    # Get the Submitty repo path
+    # script_path = os.path.dirname(os.path.realpath(__file__))
+    repository_path = "__INSTALL__FILLIN__SUBMITTY_REPOSITORY__"
+    script_path = os.path.join("__INSTALL__FILLIN__SUBMITTY_INSTALL_DIR__","test_suite","rainbowGrades")
+    runner_dir = os.path.join("__INSTALL__FILLIN__SUBMITTY_DATA_DIR__","to_be_graded_batch")
+
+    # Verify resources exist, set up initial temporary directories and configuration
+    print("Creating temporary RainbowGrades test directories")
+    test_tmp = tempfile.mkdtemp("", "")
+    print("Made new directory {}".format(test_tmp))
+
+    if not os.path.exists(test_tmp):
+        error_and_cleanup(test_tmp, "Failed to create temporary directory")
+
+    for f in os.listdir(runner_dir):
+        if "__csci1100__" in f:
+            error_and_cleanup(test_tmp,"csci1100 has assignments in the grading queue."
+                                       " Wait for the autograder to finish and then generate new grade summary reports"
+                                       "prior to re-running this test.")
+
+    rainbow_path = os.path.join(repository_path,"RainbowGrades")
+    if not os.path.exists(rainbow_path):
+        error_and_cleanup(test_tmp, "Couldn't find Rainbow Grades source code")
+
+    rainbow_tmp = os.path.join(test_tmp, "rainbow_grades")
+    os.mkdir(rainbow_tmp)
+
+    summary_tmp = os.path.join(test_tmp, "grade_summaries")
+    os.mkdir(summary_tmp)
+
+    grading_tmp = os.path.join(test_tmp, "grading")
+    os.mkdir(grading_tmp)
+
+    if not os.path.exists(rainbow_tmp) or not os.path.exists(summary_tmp):
```

Comment:
Yeah, there's not really a reason not to and does make us extra sure that what we're looking at is a directory (and use `isfile` if looking at a file).